integration-tests/src/main/java/org/apache/polaris/service/it/ext/PolarisIntegrationTestExtension.java

```diff
@@ -0,0 +1,157 @@
+      ServiceLoader.load(PolarisServerManager.class)
+          .findFirst()
+          .orElseThrow(() -> new IllegalStateException("PolarisServerManager not found"));
+
+  @Override
+  public boolean supportsParameter(
+      ParameterContext parameterContext, ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+    Class<?> type = parameterContext.getParameter().getType();
+    return type.isAssignableFrom(PolarisApiClient.class)
+        || type.isAssignableFrom(ClientCredentials.class)
+        || type.isAssignableFrom(AuthToken.class);
+  }
+
+  @Override
+  public Object resolveParameter(
+      ParameterContext parameterContext, ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+    Env env = env(extensionContext);
+    Class<?> type = parameterContext.getParameter().getType();
+    if (type.isAssignableFrom(PolarisApiClient.class)) {
+      return env.client();
+    } else if (type.isAssignableFrom(ClientCredentials.class)) {
+      return env.server.adminCredentials();
+    } else if (type.isAssignableFrom(AuthToken.class)) {
+      return env.server.adminToken();
+    }
+    throw new IllegalStateException("Unable to resolve parameter: " + parameterContext);
+  }
+
+  private Env env(ExtensionContext context) {
+    ExtensionContext classCtx = classContext(context);
+    ExtensionContext.Store store = classCtx.getStore(NAMESPACE);
+    return store.getOrComputeIfAbsent(
+        Env.class,
+        (key) -> {
+          Class<?> testClass = classCtx.getRequiredTestClass();
+          Map<String, String> featureConfig =
+              AnnotationUtils.findRepeatableAnnotations(testClass, PolarisFeatureConfig.class)
+                  .stream()
+                  .collect(
+                      Collectors.toMap(PolarisFeatureConfig::name, PolarisFeatureConfig::value));
+          return new Env(manager.realm(), manager.startServer(featureConfig));
+        },
+        Env.class);
+  }
+
+  private ExtensionContext classContext(ExtensionContext context) {
+    while (context.getParent().isPresent()) {
+      UniqueId id = UniqueId.parse(context.getUniqueId());
+      if ("class".equals(id.getLastSegment().getType())) {
+        break;
+      }
+
+      context = context.getParent().get();
+    }
+
+    return context;
+  }
+
+  private static class Env implements CloseableResource {
+    private final String realm;
+    private final Server server;
+    private final Client client;
+
+    private Env(String realm, Server server) {
+      this.realm = realm;
+      this.server = server;
+
+      ObjectMapper mapper = new ObjectMapper();
+      mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+      mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+      mapper.setPropertyNamingStrategy(new PropertyNamingStrategies.KebabCaseStrategy());
+      RESTSerializers.registerAll(mapper);
```

Comment:
I think we are missing a call to:

```java
org.apache.polaris.service.config.Serializers.registerSerializers(objectMapper)
```

Reply:
It does not seem necessary at this point :shrug: It deals only with _de-_serialization ATM.